### DISCUSSION TITLE
images shown in slices by ximtool

### DISCUSSION THREAD

**Discussion by marclg83** [iraf-community/iraf#414] [Q&A]
I recently installed the X11IRAF using the installer (Intel 64 bit) in a Mac laptop with Big Sur 11.7.8. IRAF works great (i.e., apall, imcombine, imarith, splot...), however, the ximtool displays only a section of 512x512 pixels if I attempt to use the imexam or display tasks. 

I looked at the ximtool help and found that a different size can be defined  by creating an .imtoolrc file in my home directory. Following the instructions I create this file and add the following line inside:

1  2  2048  2048

As a result, when I display an 2048x2048-pixel image, the tool shows slices of the image as the attached screenshot. I tried with different sizes in the .imtoolrc file, but it is always the same result with the exception of the 512x512 size, which shows a clipped version of the image.

Might you have any clue on what's going on here? I guess it's a stupid mistake, but I cannot figure it out. Thank you very much in advance for any help that you could provide!

<img width="988" alt="Screenshot 2024-10-30 at 10 23 20" src="https://github.com/user-attachments/assets/6d2b3062-ad14-4f11-99fc-092753b5257a">

**Comment by olebole** [comment]
Creating a local imtoolrc is usually unnecessary and dangerous, because its entries must match the assumtions IRAF has about the device. Specifically your single line makes the first configuration of ximtool as a 2048x2048 image, while IRAF still assumes a 512x512 image, which results in the image you ave shown.
The default imtoolrc already has several image buffers pre-defined, so there is no need to re-define them. It is enough to set the **stdimage** variable in IRAF to the 2048 one:
```
ecl> set stdgraph=imt2048
ecl> imexam myimg 1
```

However, the use of [SAOImageDS9](https://ds9.si.edu) may be the better choice, as this program is much more flexible than ximtool.

As this is not a bug, I moved it into the discussion forum.

Finally, if you are just going to start using IRAF: It is not recommended to start new projects using IRAF nowadays, unless it is really unavoidable IRAF is an old package without institutional support and a shrinking user and maintainer base. Instead, it is really better to use f.e. the [Astropy](https://astropy.org) ecosystem which is already a good alternative for many use cases instead of investing time into a depcreated software tool.

**Reply by lesliehebb** [reply]
I had a similar problem where the default display of the image was only showing the middle 512x512 pixels, but to fix it I needed to set the stdimage=imt2048, not the stdgraph (as shown above).

ecl> show stdgraph
ecl> show stdimage
ecl> displ myimg 1
ecl > imexam 

<img width="1162" height="832" alt="screenshot_ 2026-02-24 at 12 39 46 PM" src="https://github.com/user-attachments/assets/37e4afe3-2e39-415a-9cfb-9254cfb96355" />

ecl> set stdimage=imt2048
ecl> show stdimage
ecl> displ myimg 1
ecl > imexam 

<img width="1162" height="836" alt="screenshot_ 2026-02-24 at 12 40 28 PM" src="https://github.com/user-attachments/assets/23e68c09-22f8-4fd7-afdd-e1dc927fffb3" />
